src/components/MyAccount/RequestsTab/UpdateLocation.tsx

```diff
@@ -30,88 +30,99 @@ const UpdateLocation = ({
   pickupLocationOptions,
   patronId,
   hold,
-  pickupLocation,
   key,
 }: UpdateLocationPropsType) => {
+  const selectRef = useRef(null)
+  const updateLocationButtonRef = useRef(null)
+
+  const [focusOnChangeLocationButton, setFocusOnChangeLocationButton] =
+    useState(false)
+
   const { getMostUpdatedSierraAccountData } = useContext(PatronDataContext)
   const { Modal, onOpen: openModal, onClose: closeModal } = useModal()
+
   const locationsWithSelectedFirst = [
-    pickupLocation,
+    hold.pickupLocation,
```

Comment:
Just a thought about why the new selected location is not showing up, if `pickupLocation` is in an object, React won't pick that up as a new update since it doesn't compare nested object values that well. Not sure if that is exactly why you ran into the bug but worth investing that in the next pr.